onecodex/input_helpers.py

```diff
@@ -3,51 +3,140 @@
 import os
 import shutil
 import tempfile
+import logging
 from collections import defaultdict
 
+# Captures parts before and after ordinal
+# (. or _ followed by num followed by . or _ and non-digits)
+ORDINAL_REV_PATTERN = r"([._])\d([._][\D._]+)$"
+
+# Captures parts before and after paired ordinal (as above, but includes R1, r1, R2, r2)
+PAIRED_ORDINAL_REV_PATTERN = r"([._][Rr])\d+([._][\w.]+)$"
+
+log = logging.getLogger("onecodex")
+
+
+def _replace_filename_ordinal(filename, replacement):
+    """Replace file[_.]num[._] with file[_.]replacement[._]."""
+    replace_pattern = rf"\g<1>{replacement}\g<2>"
+    return re.sub(ORDINAL_REV_PATTERN, replace_pattern, filename)
+
+
+def _replace_paired_filename_ordinal(filename, replacement):
+    """Replace file[_.][Rr]?num[._] with file[_.][Rr]?replacement[._]."""
+    first_pass = _replace_filename_ordinal(filename, replacement)
+    replace_pattern = rf"\g<1>{replacement}\g<2>"
+    return re.sub(PAIRED_ORDINAL_REV_PATTERN, replace_pattern, first_pass)
+
+
+def concatenate_ont_groups(files, prompt):
+    """Concatenate ONT split files and return the group as a single entry on the files list."""
+    single_files = set(files)
+    ont_groups = defaultdict(set)
+    auto_group = True
+
+    for filename in files:
+        ont_zero_filename = _replace_filename_ordinal(filename, "0")
+        if os.path.exists(ont_zero_filename):
+            # strip the ordinal and preceding . or _
+            base_filename = re.sub(r"[._]\d([._][\D._]+)$", r"\1", filename)
+            base_filename = os.path.join(tempfile.gettempdir(), base_filename)
```

Comment:
It'd be nice to clean up the temporary concatenated files when the process completes

Reply:
Due to using `tempdir`, the file is already being removed. I've just checked and the concatenated file is not on the filesystem after the command finishes.

Reply:
Good catch, I thought it was being cleaned up already!